test/js/search/search-query-display.test.js

```diff
@@ -1,61 +1,94 @@
     const wrapper = mount((
       <SearchQueryDisplay facets={oneStringFacet} terms={''}/>
     ))
-    expect(wrapper.text().trim()).toEqual(': Metadata contains (species: Homo sapiens)')
+    expect(wrapper.text().trim()).toEqual(': Metadata contains (species: Homo sapiens) Clear All')
   })
 
   it('renders multiple facets', async () => {
     const wrapper = mount((
       <SearchQueryDisplay facets={twoStringFacets} terms={''}/>
     ))
-    expect(wrapper.text().trim()).toEqual(': Metadata contains (disease: disease1) AND (species: Homo sapiens)')
+    expect(wrapper.text().trim()).toEqual(': Metadata contains (disease: disease1) AND (species: Homo sapiens) Clear All')
   })
 
   it('renders string and numeric facets', async () => {
     const wrapper = mount((
       <SearchQueryDisplay facets={stringAndNumericFacets} terms={''}/>
     ))
-    expect(wrapper.text().trim()).toEqual(': Metadata contains (species: Homo sapiens OR Mus musculus) AND (organism age: 14 - 180 years)')
+    expect(wrapper.text().trim()).toEqual(': Metadata contains (species: Homo sapiens OR Mus musculus) AND (organism age: 14 - 180 years) Clear All')
   })
 
   it('renders terms', async () => {
     const wrapper = mount((
       <SearchQueryDisplay facets={[]} terms={['foo']}/>
     ))
-    expect(wrapper.text().trim()).toEqual(': Text contains (foo)')
+    expect(wrapper.text().trim()).toEqual(': Text contains (foo) Clear All')
   })
 
   it('renders terms and a single facet', async () => {
     const wrapper = mount((
       <SearchQueryDisplay facets={oneStringFacet} terms={['foo', 'bar']}/>
     ))
-    expect(wrapper.text().trim()).toEqual(': (Text contains (foo OR bar)) AND (Metadata contains (species: Homo sapiens))')
+    expect(wrapper.text().trim()).toEqual(': (Text contains (foo OR bar)) AND (Metadata contains (species: Homo sapiens)) Clear All')
+  })
+})
+
+describe('Clearing search query', () => {
+  it('clears search params', () => {
+    const speciesFacet = {
+      name: 'Species',
+      id: 'species',
+      type: 'string',
+      links: [{ name: 'NCBI Taxonomy', url: 'https://foo.tdb' }],
+      filters: [
+        { id: 'NCBITaxon_9606', name: 'Homo Sapiens' }
+      ],
+      links: []
+    }
+    const component = <PropsStudySearchProvider searchParams={{ terms: 'foo', facets: { species: ['NCBITaxon_9606'] } }}>
+      <ClearAllButton/>
+      <KeywordSearch/>
+      <FacetControl facet={speciesFacet}/>
+    </PropsStudySearchProvider>
+    const wrapper = mount(component)
+    wrapper.find('#facet-species > a').simulate('click')
+    wrapper.find(ClearAllButton).simulate('click')
+    expect(wrapper.find('input[name="keywordText"]').first().props().value).toEqual('')
+    // Check if badge for filter doesn't exist
+    expect(wrapper.find('.filter-badge-list')).toHaveLength(0)
+    // Filter should not be checked
+    expect(wrapper.find('input[name="NCBITaxon_9606"]').props().checked).toEqual(false)
```

Comment:
great assertions.  I would duplicate them all before the `wrapper.find(ClearAllButton).simulate('click')` line so that you're both checking that the values were there beforehand, and they get cleared afterwards